src/shared/acpi-fpdt.c

```diff
@@ -61,6 +61,34 @@ struct acpi_fpdt_boot {
         uint64_t exit_services_exit;
 } _packed;
 
+/* /dev/mem is deprecated on many systems try using /sys/firmware/acpi/fpdt parsing instead */
+static int acp_get_boot_usec_alt(usec_t *loader_start, usec_t *loader_exit) {
+        usec_t ret;
+        int r;
+
+        r = read_timestamp_file("/sys/firmware/acpi/fpdt/boot/exitbootservice_end_ns", &ret);
+        if (r < 0)
+                return -errno;
```

Comment:
error is returned in `r`

Reply:
Yes, that makes more sense, it wasn't clear why the original code was doing it that way (messing with the errno).

Reply:
This should be done